upath/core.py

```diff
@@ -136,50 +139,66 @@ class UPath(pathlib.Path):
     _closed: bool
     _accessor: _FSSpecAccessor
 
+    @classmethod
+    def _from_path(
```

Comment:
I don't think it's worth extracting this as a new classmethod.
I agree that `__new__` is a mess, but at least it's contained in that one method.

Reply:
I think `__new__` really benefits from the simplification, so I moved the function outside of the class so that it does not clutter the API.  If this is still undesirable, I will move it back into `__new__`, it just seems like a significant improvement in readability.